server/src/test/java/org/opensearch/repositories/blobstore/BlobStoreRepositoryRemoteIndexTests.java

```diff
@@ -368,4 +378,239 @@ public void testRetrieveShallowCopySnapshotCase2() throws IOException {
+
+        lockFiles = getLockFilesInRemoteStore(remoteStoreIndexName, remoteStoreRepositoryName);
+        assert (lockFiles.length == 1) : "there should be only one lock file, but found " + Arrays.toString(lockFiles);
+        assert lockFiles[0].endsWith(snapshotId1.getUUID() + ".lock");
+
+        logger.info("--> make sure the node's repository can resolve the snapshots");
+        final List<SnapshotId> originalSnapshots = Arrays.asList(snapshotId1, snapshotId3);
+
+        final RepositoriesService repositoriesService = getInstanceFromNode(RepositoriesService.class);
+        final BlobStoreRepository repository = (BlobStoreRepository) repositoriesService.repository(snapshotRepositoryName);
+        RepositoryData repositoryData = OpenSearchBlobStoreRepositoryIntegTestCase.getRepositoryData(repository);
+        IndexId indexId = repositoryData.resolveIndexId(remoteStoreIndexName);
+
+        List<SnapshotId> snapshotIds = repositoryData.getSnapshotIds()
+            .stream()
+            .sorted((s1, s2) -> s1.getName().compareTo(s2.getName()))
+            .collect(Collectors.toList());
+        assertThat(snapshotIds, equalTo(originalSnapshots));
+    }
+
+    public void testLocksForShallowClonePostDelete() throws IOException {
+        FeatureFlagSetter.set(FeatureFlags.REMOTE_STORE);
+        final Client client = client();
+        final String snapshotRepositoryName = "test-repo";
+        final String remoteStoreRepositoryName = "test-rs-repo";
+
+        logger.info("-->  creating snapshot repository");
+
+        Settings snapshotRepoSettings = Settings.builder()
+            .put(node().settings())
+            .put("location", OpenSearchIntegTestCase.randomRepoPath(node().settings()))
+            .put(BlobStoreRepository.REMOTE_STORE_INDEX_SHALLOW_COPY.getKey(), Boolean.TRUE)
+            .build();
+        createRepository(client, snapshotRepositoryName, snapshotRepoSettings);
+
+        logger.info("-->  creating remote store repository");
+        Settings remoteStoreRepoSettings = Settings.builder()
+            .put(node().settings())
+            .put("location", OpenSearchIntegTestCase.randomRepoPath(node().settings()))
+            .build();
+        createRepository(client, remoteStoreRepositoryName, remoteStoreRepoSettings);
+
+        logger.info("--> creating an index and indexing documents");
+        final String indexName = "test-idx";
+        createIndex(indexName);
+        ensureGreen();
+        indexDocuments(client, indexName);
+
+        logger.info("--> creating a remote store enabled index and indexing documents");
+        final String remoteStoreIndexName = "test-rs-idx";
+        Settings indexSettings = getRemoteStoreBackedIndexSettings(remoteStoreRepositoryName);
+        createIndex(remoteStoreIndexName, indexSettings);
+        indexDocuments(client, remoteStoreIndexName);
+
+        deleteExtraFSFilesIfAny(remoteStoreIndexName, remoteStoreRepositoryName);
```

Comment:
these files gets created in the test directory, we don't expect these random files in user's store. to avoid these failures without cleaning up, maybe we can update our code to list files using prefix instead of using `listAll()` call.

Reply:
Yes, In lockManager it should filter out the lock files based on suffix to use only the lock files and ignore any other files in the directory.